packages/react-pages/src/client/state.ts

```diff
@@ -94,31 +94,26 @@ if (import.meta.hot) {
     }
   })
 
-  const dataPathAtoms = atomFamily((path: string) => (get) => {
+  const dataAtoms = atomFamily((path: string) => (get) => {
     const pages = get(pagesAtom)
-    const page = pages[path] || pages['/404']
-    return page?.dataPath || null
+    return pages[path]
```

Comment:
This isn't right. We must return a primitive value here (eg: the `dataPath`), or else these atoms will re-render dependent components even if the page's `dataPath` did not change, since the import function will be replaced every time. This is why the pages module needs to export `dataPath` instead of import function in development.

Reply:
pages[path] is a stable object reference, too. I don't think it will trigger unnecessary re-render.

Reply:
It's *not* a stable reference, because the `/@react-pages/pages` module is re-executed whenever any page changes. 
See here:
https://github.com/vitejs/vite-plugin-react-pages/blob/2c2dd3c6fab76ef722be538eea6cfa45d3b1aabb/packages/react-pages/src/client/state.ts#L75

Reply:
You have a point. But we also can't use `import(dataPath)` becasue vite hmr will not work. (the import chain can't be recognised by vite)

Reply:
When page file is edited and hmr happens, I think it is ok to have `/@react-pages/pages` re-executed and update all page data's object reference. After all, only one page is mounted at that time and only it can get re-render!

Reply:
There's a better way. See my comment here: https://github.com/vitejs/vite-plugin-react-pages/pull/19#discussion_r602745661

Reply:
Fixed in [this commit](https://github.com/vitejs/vite-plugin-react-pages/pull/19/commits/e9911c246559178d43e1344dcf6941dd3ea0bd2c).